server/core/src/irods_server_control_plane.cpp

```diff
@@ -1111,53 +1149,47 @@ namespace irods {
         if ( !ret.ok() ) {
             irods::log( PASS( ret ) );
             return PASS( ret );
-
         }
 
-
         auto in = avro::memoryInputStream(
                 static_cast<const uint8_t*>(
                     data_to_process.data() ),
                 data_to_process.size() );
         avro::DecoderPtr dec = avro::binaryDecoder();
         dec->init( *in );
 
-        control_plane_command cmd;
-        avro::decode( *dec, cmd );
-
-        std::string cmd_name, cmd_option, wait_option;
-        host_list_t cmd_hosts;
-        size_t wait_seconds = 0;
-        ret = extract_command_parameters(
-                  cmd,
-                  cmd_name,
-                  cmd_option,
-                  wait_option,
-                  wait_seconds,
-                  cmd_hosts );
-        if ( !ret.ok() ) {
-            irods::log( PASS( ret ) );
-            return PASS( ret );
+        avro::decode( *dec, _cmd );
 
-        }
+        return SUCCESS();
+    } // decrypt_incoming_command
+
+    error server_control_executor::perform_operation(
+            const std::string& _cmd_name,
+            const std::string& _cmd_option,
+            const std::string& _wait_option,
+            const size_t&      _wait_seconds,
+            const host_list_t& _cmd_hosts,
+            std::string&       _output ) {
+        error final_ret = SUCCESS();
+        host_list_t cmd_hosts = _cmd_hosts;
 
         // add safeguards - if server is paused only allow a resume call
         server_state& s = server_state::instance();
         std::string the_server_state = s();
         if ( server_state::PAUSED == the_server_state &&
-                SERVER_CONTROL_RESUME != cmd_name ) {
+                SERVER_CONTROL_RESUME != _cmd_name ) {
             _output = SERVER_PAUSED_ERROR;
             return SUCCESS();
         }
 
         // the icat needs to be notified first in certain
         // cases such as RESUME where it is needed to capture
         // the host list for validation, etc
-        ret = notify_icat_and_local_servers_preop(
-                  cmd_name,
-                  cmd_option,
-                  wait_option,
-                  wait_seconds,
+        irods::error ret = notify_icat_and_local_servers_preop(
```

Comment:
consider a rename here ... from "icat" to "provider"

Reply:
my PR takes care of this so renames can be ignored for now if we agree.

Reply:
ah, very nice.   didn't realize it was the same code.